### DISCUSSION TITLE
Part screen snapshot

### DISCUSSION THREAD

**Discussion by horan74** [dlemstra/Magick.NET#702] [Help] [answered]
Hello!

I created a little class to take screenshots and convert them to opencv's Mat objects

```
using ImageMagick;
using OpenCvSharp;
using System.IO;
using System;

class ScreenShoter {
	private static MagickReadSettings settings = new MagickReadSettings();
	private static MemoryStream stream = new MemoryStream();
	private static MagickImage image = new MagickImage();

	public static Mat ScreenShot(ImreadModes readMode) {
		
		try {
		  image.Read("SCREENSHOT", MagickFormat.Screenshot);
   		  image.Format = MagickFormat.Bmp;
    		  image.Write(stream);
		} catch (Exception ex) {
		  Console.WriteLine(ex);
		}
    	stream.Position = 0;
    	Mat result = Mat.FromStream(stream, readMode);
    	stream.SetLength(0);

        return result;
	}
}
```

I'm gonna use the static method "ScreenShot" in endless while cycle.

The method

`image.Read("SCREENSHOT", MagickFormat.Screenshot);`

takes snapshot from the entire monitor. I know that after taking the snapshot I can crop it! But taking a snapshot from entire screen is a slow operation.

Is there the way to take a part of the screen using Magick.NET api?

**Comment by dlemstra** [comment, marked as answer]
Currently you can crop the image automatically like this:

```C#
var settings = new MagickReadSettings
{
    ExtractArea = new MagickGeometry(20, 30, 500, 400)
};

using (var image = new MagickImage("screenshot:", settings))
{
}

// OR

using (var image = new MagickImage("screenshot:[500x400+20+30]"))
{
}
```

But this now takes a screenshot of the whole area instead of part of the screen. I just pushed a patch to impove the performance and only copy the pixels of the requested area. Not sure yet if this is a solid solution and it will need some more testing but I hope that this will become available in the next release.

**Reply by horan74** [reply]
Thank you very much! You really helped me!

I have added the second method in my class:

```
public static Mat ScreenShot(int x, int y, int width, int height, ImreadModes readMode) {
	
	try {
		settings.ExtractArea.X = x;
		settings.ExtractArea.Y = y;
		settings.ExtractArea.Width = width;
		settings.ExtractArea.Height = height;
		image.Read("screenshot:", settings);
		image.Format = MagickFormat.Bmp;
		image.Write(stream);
	} catch (Exception ex) {
		Console.WriteLine(ex);
	}
	stream.Position = 0;
	Mat result = Mat.FromStream(stream, readMode);
	stream.SetLength(0);
        return result;
}
```
I have measured the time(Stopwatch class) taking the calls of both methods. The second one takes less time than the first one!

**Reply by dlemstra** [reply]
The patch to also only capture that specific region of the screen will become available in the next release. This should give you a minor performance improvement.
